scipy/sparse/linalg/_eigen/tests/test_svds.py

```diff
@@ -289,22 +291,15 @@ def test_svds_parameter_tol(self):
         A = csc_matrix(A)
 
         def err(tol):
-            if self.solver == 'lobpcg' and tol == 1e-4:
-                msg = ("Exited at iteration.*|"
-                       "Exited postprocessing with accuracies.*")
-                with pytest.warns(UserWarning, match=msg):
-                    _, s2, _ = svds(A, k=k, v0=np.ones(n),
-                                    solver=self.solver, tol=tol)
-            else:
-                _, s2, _ = svds(A, k=k, v0=np.ones(n),
-                                solver=self.solver, tol=tol)
+            _, s2, _ = svds(A, k=k, v0=np.ones(n),
+                            solver=self.solver, tol=tol)
             return np.linalg.norm((s2 - s[k-1::-1])/s[k-1::-1])
 
         tols = [1e-4, 1e-2, 1e0]  # tolerance levels to check
         # for 'arpack' and 'propack', accuracies make discrete steps
         accuracies = {'propack': [1e-12, 1e-6, 1e-4],
                       'arpack': [2.5e-15, 1e-10, 1e-10],
-                      'lobpcg': [1e-10, 1e-3, 10]}
+                      'lobpcg': [1e-6, 0.1, 10]}
```

Comment:
If one has to drop the accuracy so badly, from 1e-10 to 1e-6, in order to pass the test, it may be an indication of a serious issue with accuracy in one of the libraries called - looks like a bug giving a single precision accuracy instead of double precision at least in one of the library functions.  

Probably a proper solution is to file a bug report while maybe temporarily lowing the accuracy to avoid annoying build error messages.

Reply:
Thanks for the suggestion! Just wanted to note that this particular test was disabled by gh-11829 two years ago. In gh-19855, I re-enabled this test by deleting the early return statement, exposing the accuracy issue. Could you help me open a separate issue for this? I would do it, but I don't know anything at all about sparse linear algebra.

Reply:
No specific knowledge is needed here. If the same test consistently passes with one set of libraries but also consistently fails with a different set of libraries, there is something wrong with the latter set of libraries. If you can single out the test and have examples of failures like your
<html>
<body>
<!--StartFragment--><h5 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 0.875em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Test_SVDS_LOBPCG.test_svds_parameter_tol (10,000 total)</h5>
Machine | 3444b2d | 2bfd351
-- | -- | --
M2 macOS 14 | 9938 | 0
x64 Ubuntu 22.04 | 9944 | 0

<!--EndFragment-->
</body>
</html>Test_SVDS_LOBPCG.test_svds_parameter_tol (10,000 total)
Machine	https://github.com/scipy/scipy/commit/3444b2d5388b6fb81524c96b7971c49393a745fb	https://github.com/scipy/scipy/commit/2bfd351321ad6706b228f2c9959e56c96431c03b
M2 macOS 14	9938	0
x64 Ubuntu 22.04	9944	0 

you just need to add examples of passes for the same test with different libraries to claim a bug in the libraries. 

If the test fails with all the libraries, the bug is in the code. Since this particular test was disabled by https://github.com/scipy/scipy/pull/11829 it would more likely be a bug in the svds so can be added to already rather long list https://github.com/scipy/scipy/issues?q=is%3Aissue+is%3Aopen+svds ....

Reply:
I reran 10,000 iterations of `Test_SVDS_LOBPCG.test_svds_parameter_tol` on 3444b2d and now only see ~3200 failures. Not sure how I got ~9900 before. I was able to reproduce the reported failure rates for the other tests. Sorry for the confusion! I'll update the table to reflect this.